docs/better-sandwiches/index.md

```diff
@@ -0,0 +1,310 @@
+artisanal mushrooms lovingly grown by mustachioed hipsters on a local farm,
+which increases the price of portabello sandwiches to $10. Now portabello and
+ham sandwiches are the same price, so you update the `Item.toPrice` function
+accordingly:
+
+<<< Item.re#to-price-same-price{3}
+
+We can combine the portabello and ham branches because they return the same
+value:
+
+<<< Item.re#to-price-combine{3-4}
+
+Since the outer constructor `Sandwich` is the same in both branches, we can
+combine them into a single branch:
+
+<<< Item.re#to-price{3}
+
+The power of pattern matching gives us both clarity and brevity. This is the
+OCaml Way!
+
+## Update `Item.toEmoji` function
+
+`Item.toEmoji` should now return a combination of two emojis when it gets a
+`Sandwich(_)` argument:
+
+| Sandwich | Emoji |
+| -------- | ----- |
+| Portabello | 🥪(🍄) |
+| Ham | 🥪(🐷) |
+| Unicorn | 🥪(🦄) |
+
+We can implement this using `Js.Array.joinWith`:
+
+<<< Item.re#to-emoji-joinwith{3-13}
+
+This touches on the issue of coding style for switch expressions. When possible,
+put shorter branches before longer branches:
+
+<<< Item.re#to-emoji-short-branch-first
+
+## `{j||j}` quoted string literal
+
+The approach using `Js.Array.joinWith` works, but readability could be improved
+by using a [quoted string literal with the quoted string identifier
+`j`](https://melange.re/v2.2.0/communicate-with-javascript/#strings):
+
+<<< Item.re#to-emoji-j-string-literal{5-13}
+
+`{j||j}` quoted string literals are similar to [template literals in
+JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
+except that they can only accept variables, not arbitrary expressions. Variables
+that appear in `{j||j}` quoted string literals are implicitly converted to
+strings, so they are not type safe and thus unsuitable for production code. See
+the [exercises at the end of this chapter](#exercises) for more details.
+
+Also note that unlike switch expressions, the `fun` syntax does not accept
+multi-line expressions in branches unless you add `{}` around them.
+
+::: warning
+
+If you need to make a string literal that contains Unicode characters, use
+`{js||js}`. You may be tempted to use `{j||j}` because it also supports Unicode
+characters and it saves some keystrokes, but as explained above, it isn't
+type-safe and is therefore less future-proof.
+
+:::
+
+## `Printf.sprintf`
+
+The OCaml standard library provides a type-safe way to do string interpolation
+in the form of the
+[Printf.sprintf](https://melange.re/v2.2.0/api/re/melange/Stdlib/Printf/index.html#val-sprintf)
+function:
+
+<<< Item.re#to-emoji-sprintf
+
+Unlike `{j||j}` quoted string literals, `Printf.sprintf` will not do implicit
+string conversion.
+
+::: info
+
+Be aware that using functions from the `Printf` module can add a surprising
+amount to your app's production bundle size. In the app from this chapter, the
+size of `main.js` generated by Webpack went from 144.2 kB (using
```

Comment:
Maybe worth mentioning if these measurements are minified / uglified and gzipped or not. E.g. 

> 144.2 kB (unminified, not compressed)

Reply:
Indicate that the numbers are uncompressed in https://github.com/melange-re/melange-for-react-devs/pull/13/commits/036768e9de50ae4188a1bd8eb5ff4b1ceb3ed49a